IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m

```diff
@@ -233,8 +234,15 @@ - (nullable NSData *)decrypt:(nonnull NSString *)encryptedMessageString
 
 - (NSString *)signData:(NSString *)message
 {
-    NSData *signedData = [[message dataUsingEncoding:NSUTF8StringEncoding] msidSignHashWithPrivateKey:self.privateKeyRef];
-    return [[NSString alloc] initWithData:signedData encoding:NSUTF8StringEncoding];
+    if ([message length] == 0)
+    {
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Message to sign was empty");
+        return nil;
+    }
+    
+    NSData *hashedData = [[message dataUsingEncoding:NSUTF8StringEncoding] msidSHA256];
+    NSData *signedData = [hashedData msidSignHashWithPrivateKey:self.privateKeyRef];
+    return [signedData msidHexString];
```

Comment:
Why does it need to be hex string? Will it potentially break other callers?

Reply:
I wanted to discuss this with you as well. I think we should return a base64 string as signature cannot be shared over the internet in binary format.

Reply:
Will update the code.